examples/02_decoding/plot_miyawaki_encoding.py

```diff
@@ -0,0 +1,233 @@
+"""
+============================================================
+Encoding models for visual stimuli from Miyawaki et al. 2008
+============================================================
+
+This example demonstrates how to build an encoding model for
+functional MRI data with visual stimuli.
+
+Encoding models try to predict neuronal activity using properties of
+a real world stimulus, like images or sound.
+
+In the dataset from :func:`nilearn.datasets.fetch_miyawaki2008`,
+participants were shown images consisting of 10x10
+binary (either black or white) pixels and the corresponding  fMRI activity
+was recorded. We will predict the neuronal activity from the pixel-values
+of the presented images for each voxel. Then we extract the receptive fields
+for a set of voxels to see which pixel location a voxel is most sensitive to.
+
+See also :doc:`plot_miyawaki_reconstruction` for a decoding
+approach for the same dataset.
+"""
+
+##############################################################################
+# Loading the data
+# ----------------
+# First some imports:
+
+import numpy as np
+import pylab as plt
+from nilearn.plotting import plot_stat_map
+from nilearn import masking
+
+
+##############################################################################
+# Now we can load the data set:
+from nilearn.datasets import fetch_miyawaki2008
+
+dataset = fetch_miyawaki2008()
+
+##############################################################################
+# We only use the training data of this study,
+# where random binary images were shown.
+
+X_random_filenames = dataset.func[12:]
+y_random_filenames = dataset.label[12:]
+
+# shape of the binary image in pixels
+y_shape = (10, 10)
+
+
+##############################################################################
+# We can use :func:`nilearn.input_data.MultiNiftiMaser` to load the fMRI data,
+# clean and mask it.
+
+from nilearn.input_data import MultiNiftiMasker
+
+masker = MultiNiftiMasker(mask_img=dataset.mask, detrend=True,
+                          standardize=True)
+masker.fit()
+X_train = masker.transform(X_random_filenames)
+
+# We load the visual stimuli from csv files
+y_train = []
+for y in y_random_filenames:
+    y_train.append(np.reshape(np.loadtxt(y, dtype=np.int, delimiter=','),
+                              (-1,) + y_shape, order='F'))
+
```

Comment:
Ah, maybe here is where we want to plot stimuli.

I would suggest moving the definition of y_shape here, to have things close to where they are understandable.